### PR TITLE
LPS-130919 Move configure Session Property Vocabularies from System Settings to Instance Settings

### PR DESCRIPTION
**Pull request by YurenaC**
Motivation
=======

Currently, segmentation settings exist only at System level, so there’s no way to configure the feature at the Instance level.
In order to facilitate the cloud-based products development, we need to make it available at the instance level.

[Link to story or ticket](https://issues.liferay.com/browse/LPS-130919)

Proposed Solution
========
To easily solve this problem, the scope of the Session Property Vocabularies has been set to COMPANY, so after a few simple changes to allow multiple assetVocabularies, with different entityFields assigned depending on its level (system or instance) it works as expected

Steps to Verify:
----------------

1. Start the portal and go to "Control Panel > System Settings > In Content and Data select Segments"
2. In "Virtual Instance Scope" select "Session Property Vocabularies"
3. Select "add" and choose "browser" in "Session Property Name" and "Audience" in Vocabulary Name, click Save button 
4. **Check point**: Now you can see "browser" listed in "Configuration Entries"

5. Go to "Control Panel > Instance Settings > In Content and Data select Segments"
6. In "Virtual Instance Scope" select "Session Property Vocabularies"
7. Select "add" and choose "browser" in "Session Property Name" and "Stage" in Vocabulary Name, click Save button 
8. All should work as expected.

9. Repeat step 7 and a Duplication Error message should be shown. 

Change log
-----------------------

- LPS-130919 Update SegmentsContextVocabularyConfiguration scope to Company
- LPS-130919 Allow overwrite session propertes (defined at system level) to a different vocabulary at instance level

### CODE REVIEW

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
Hi @cgoncas, 
I think this changes are the simplest way to achieve our goal

Let me know your thoughts! :)

**Comment by cgoncas** [comment]
Hey @YurenaC !!

It is not posible to keep sustenta and instance settings at the same time?

thanks!!

**Comment by cgoncas** [comment]
ci:forward

**Comment by YurenaC** [comment]
> Hey @YurenaC !!
> 
> It is not posible to keep system and instance settings at the same time? What are the problems that you face with this approach?
> 
> thanks!!

I kept system and instance settings and I think it works as expected.

It allows you to define a vocabulary and assign it to "Audience" for example, at system level.
Once that is done you cannot reassign that vocabulary at system level but you can do it at instance level.

Does it make sense? I'm happy to show you in a small demo tomorrow, will be easy to understand

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:stop

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:relevant

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -13101,6 +13101,10 @@ segmentation-enabled-description=Disable this option to prevent user segmentatio
 segmentation-enabled-name=Enable Segmentation
 segments=Segments
 segments-asah-connector-configuration-name=Segments Analytics Cloud
+segments-context-vocabulary-company-configuration-asset-vocabulary-name=Vocabulary Name
```
Comment (cgoncas) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:13104]: The text should be the same as we use before, so no new keys needs to be added.

File: modules/apps/segments/segments-context-vocabulary/src/main/java/com/liferay/segments/context/vocabulary/internal/field/customizer/SegmentsContextVocabularySegmentsFieldCustomizer.java [hunk 2]
```diff
@@ -125,10 +125,11 @@ public String getLabel(String fieldName, Locale locale) {
 	@Activate
 	@Modified
 	protected void activate(Map<String, Object> properties) {
-		SegmentsContextVocabularyConfiguration
+		SegmentsContextVocabularyCompanyConfiguration
 			segmentsContextVocabularyConfiguration =
 				ConfigurableUtil.createConfigurable(
-					SegmentsContextVocabularyConfiguration.class, properties);
```
Comment (cgoncas) [inline modules/apps/segments/segments-context-vocabulary/src/main/java/com/liferay/segments/context/vocabulary/internal/field/customizer/SegmentsContextVocabularySegmentsFieldCustomizer.java:131]: You should read both, company and system configuration.

Comment (cgoncas) [reply modules/apps/segments/segments-context-vocabulary/src/main/java/com/liferay/segments/context/vocabulary/internal/field/customizer/SegmentsContextVocabularySegmentsFieldCustomizer.java:131]: Probably you will need a Configuration Provider, the same way we have in PageAudit

**Review by cgoncas — CHANGES_REQUESTED** [review]
Hey @YurenaC !!

Do you mind add some test to these changes!!

Thanks!!

**Comment by CarlosBrichete** [comment]
ci:test:tango

**Comment by CarlosBrichete** [comment]
Hey @cgoncas I've seen that now the session property vocabularies are in Instance Settings, but they are still in System Settings too. Is this expected?

**Comment by cgoncas** [comment]
> Hey @cgoncas I've seen that now the session property vocabularies are in Instance Settings, but they are still in System Settings too. Is this expected?

Yeahh @CarlosBrichete !!

I will ask Joe to update the AC, if we delete the Configuration from System then an upgrade process or something should be done for our old customers, this way it is saffer.

Thanks!!

**Pull request merged**